sqldelight-gradle-plugin/src/main/kotlin/com/squareup/sqldelight/gradle/GenerateMigrationOutputTask.kt

```diff
@@ -0,0 +1,97 @@
+package com.squareup.sqldelight.gradle
+
+import com.squareup.sqldelight.VERSION
+import com.squareup.sqldelight.core.SqlDelightDatabaseProperties
+import com.squareup.sqldelight.core.SqlDelightEnvironment
+import com.squareup.sqldelight.core.lang.util.rawSqlText
+import java.io.File
+import javax.inject.Inject
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.FileTree
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.SourceTask
+import org.gradle.api.tasks.TaskAction
+import org.gradle.workers.WorkAction
+import org.gradle.workers.WorkParameters
+import org.gradle.workers.WorkerExecutor
+
+@CacheableTask
+abstract class GenerateMigrationOutputTask : SourceTask() {
+  @Suppress("unused") // Required to invalidate the task on version updates.
+  @Input val pluginVersion = VERSION
+
+  @get:Inject
+  abstract val workerExecutor: WorkerExecutor
+
+  @get:OutputDirectory
+  var outputDirectory: File? = null
+
+  @Internal lateinit var sourceFolders: Iterable<File>
+  @Input lateinit var properties: SqlDelightDatabaseProperties
+  @Input lateinit var migrationOutputExtension: String
+
+  @TaskAction
+  fun generateSchemaFile() {
+    workerExecutor.noIsolation().submit(GenerateSchema::class.java) {
+      it.sourceFolders.set(sourceFolders.filter(File::exists))
+      it.outputDirectory.set(outputDirectory)
+      it.moduleName.set(project.name)
+      it.propertiesJson.set(properties.toJson())
+      it.migrationExtension.set(migrationOutputExtension)
+    }
+  }
+
+  @InputFiles
+  @SkipWhenEmpty
+  @PathSensitive(PathSensitivity.RELATIVE)
+  override fun getSource(): FileTree {
+    return super.getSource()
```

Comment:
How come you don't have to implement this? Will this re-run when any source file changes? Or do we want to only re-run on `.sqm` file changes?

Reply:
yea the override is so that the annotations instructing gradle when to invalidate are there

Reply:
it gets set here: https://github.com/cashapp/sqldelight/pull/1736/files#diff-42089ce59d3d622205b502f8d59edf74R192